src/main/scala/uk/gov/nationalarchives/Lambda.scala

```diff
@@ -39,16 +44,27 @@ class Lambda extends RequestStreamHandler {
       asset <- IO.fromOption(assetItems.headOption)(
         new Exception(s"No asset found for ${input.id} and ${input.batchId}")
       )
+      fileReference = asset.identifiers.find(_.identifierName == "BornDigitalRef").map(_.value).orNull
+      log = logger.info(Map("batchRef" -> input.batchId, "fileReference" -> fileReference))(_)
       _ <- if (asset.`type` != Asset) IO.raiseError(new Exception(s"Object ${asset.id} is of type ${asset.`type`} and not 'Asset'")) else IO.unit
+      _ <- log(s"Asset ${asset.id} retrieved from Dynamo")
+
       children <- childrenOfAsset(asset, config.dynamoTableName, config.dynamoGsiName)
       _ <- IO.fromOption(children.headOption)(new Exception(s"No children found for ${input.id} and ${input.batchId}"))
+      _ <- log(s"${children.length} children found for asset ${asset.id}")
+
       _ <- children.map(child => copyFromSourceToDestination(input, config.destinationBucket, asset, child, xmlCreator)).sequence
```

Comment:
Log before starting the copy, as the copy could take some time (and maybe timeout the Lambda for larger objects)?

Reply:
I've added a line in.